src/frontend/src/expr/table_function.rs

```diff
@@ -350,6 +353,157 @@ impl TableFunction {
         }
     }
 
+    pub fn new_mysql_query(args: Vec<ExprImpl>) -> RwResult<Self> {
+        static MYSQL_ARGS_LEN: usize = 6;
+        let args = {
+            if args.len() != MYSQL_ARGS_LEN {
+                return Err(BindError("mysql_query function only accepts 6 arguments: mysql_query(hostname varchar, port varchar, username varchar, password varchar, database_name varchar, mysql_query varchar)".to_string()).into());
+            }
+            let mut cast_args = Vec::with_capacity(MYSQL_ARGS_LEN);
+            for arg in args {
+                let arg = arg.cast_implicit(DataType::Varchar)?;
+                cast_args.push(arg);
+            }
+            cast_args
+        };
+        let evaled_args = {
+            let mut evaled_args: Vec<String> = Vec::with_capacity(MYSQL_ARGS_LEN);
+            for arg in &args {
+                match arg.try_fold_const() {
+                    Some(Ok(value)) => {
+                        let Some(scalar) = value else {
+                            return Err(BindError(
+                                "mysql_query function does not accept null arguments".to_string(),
+                            )
+                            .into());
+                        };
+                        evaled_args.push(scalar.into_utf8().into());
+                    }
+                    Some(Err(err)) => {
+                        return Err(err);
+                    }
+                    None => {
+                        return Err(BindError(
+                            "mysql_query function only accepts constant arguments".to_string(),
+                        )
+                        .into());
+                    }
+                }
+            }
+            evaled_args
+        };
+
+        #[cfg(madsim)]
+        {
+            return Err(crate::error::ErrorCode::BindError(
+                "postgres_query can't be used in the madsim mode".to_string(),
+            )
+            .into());
+        }
+
+        #[cfg(not(madsim))]
+        {
+            let schema = tokio::task::block_in_place(|| {
+                RUNTIME.block_on(async {
+                    let database_opts: mysql_async::Opts = mysql_async::OptsBuilder::default()
+                        .ip_or_hostname(evaled_args[0].clone())
+                        .tcp_port(evaled_args[1].parse::<u16>().unwrap())
```

Comment:
It'd be better to check the parse result and throw an error instead of `unwrap()`.

Reply:
Fixed